blockbuster/src/com/codecool/emis/blockbuster/Blockbuster.java

```diff
@@ -0,0 +1,97 @@
+package com.codecool.emis.blockbuster;
+
+import com.codecool.emis.blockbuster.product.Product;
+
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class Blockbuster {
+
+    private Set<Product> products = new HashSet<>();
+    private Set<Product> blockBusters = new HashSet<>();
+    private int monthlyRevenu;
+    private static final int MIN_RENTED_DAY = 3;
+    private static final int MAX_RENTED_DAY = 14;
+
+
+    public void addProduct(Product product){
+        products.add(product);
+    }
+
+
+    public int averageProduct(){  // helper metod to calculate which product is better than other, I use in getBestBlockbusters()
+        int counter = 0;
+        for (Product product : products) {
+            counter +=  product.getHowManyTimesRent();
+        }
+        return counter / products.size();
+    }
+
+    public Set<Product> getBestBlockbusters(){
+        for (Product product : products) {
+            if(product.getHowManyTimesRent() > averageProduct()){
```

Comment:
you're calling the averageProduct() method again and again and again in the for loop. and that method iterates through all the products again and again. if you have tons of products, that can be a long time. 
it would be better to call the averageProduct only once before the for loop, and save its return value in a variable. that way you'll only calculate it once. 
int averageRentTime = averageProduct()
for (Product product : products) {
    if(product.getHowManyTimesRent() > averageRentTime){
    ...